packages/Webkul/Product/src/Repositories/ProductRepository.php

```diff
@@ -73,11 +73,13 @@ public function update(array $data, $id, $attribute = 'id')
 
         $product = $product->getTypeInstance()->update($data, $id, $attribute);
 
+        $updatedProduct = $this->find($id);
```

Comment:
I think $product->refresh() will do trick here. No need of again taking new variables.